core/src/Cabana_ParticleList.hpp

```diff
@@ -0,0 +1,249 @@
+
+    //! Get the underlying tuple (const).
+    KOKKOS_FORCEINLINE_FUNCTION
+    const tuple_type& tuple() const { return _tuple; }
+
+    //! The tuple this particle wraps.
+    tuple_type _tuple;
+};
+
+//---------------------------------------------------------------------------//
+//! Single SoA particle view. Wraps a view of the SoA the particle resides in.
+template <int VectorLength, class... FieldTags>
+struct ParticleView
+{
+    //! Particle AoSoA member types.
+    using traits = ParticleTraits<FieldTags...>;
+    //! Particle SoA type.
+    using soa_type = Cabana::SoA<typename traits::member_types, VectorLength>;
+    //! AoSoA vector length
+    static constexpr int vector_length = VectorLength;
+
+    //! Default constructor.
+    ParticleView() = default;
+
+    //! Tuple wrapper constructor.
+    KOKKOS_FORCEINLINE_FUNCTION
+    ParticleView( soa_type& soa, const int vector_index )
+        : _soa( soa )
+        , _vector_index( vector_index )
+    {
+    }
+
+    //! Get the underlying SoA.
+    KOKKOS_FORCEINLINE_FUNCTION
+    soa_type& soa() { return _soa; }
+
+    //! Get the underlying SoA (const).
+    KOKKOS_FORCEINLINE_FUNCTION
+    const soa_type& soa() const { return _soa; }
+
+    //! Get the vector index of the particle in the SoA.
+    KOKKOS_FORCEINLINE_FUNCTION
+    int vectorIndex() const { return _vector_index; }
+
+    //! The soa the particle is in.
+    soa_type& _soa;
+
+    //! The local vector index of the particle.
+    int _vector_index;
+};
+
+//---------------------------------------------------------------------------//
+// Particle accessors.
+//---------------------------------------------------------------------------//
+//! Return a single field for a single particle.
+template <class FieldTag, class... FieldTags, class... IndexTypes>
+KOKKOS_FORCEINLINE_FUNCTION typename std::enable_if<
+    sizeof...( IndexTypes ) == FieldTag::rank,
+    typename Particle<FieldTags...>::tuple_type::
+        template member_const_reference_type<
+            TypeIndexer<FieldTag, FieldTags...>::index>>::type
+get( const Particle<FieldTags...>& particle, FieldTag, IndexTypes... indices )
+{
+    return Cabana::get<TypeIndexer<FieldTag, FieldTags...>::index>(
+        particle.tuple(), indices... );
+}
+
+//! Return a single field for a single particle.
+template <class FieldTag, class... FieldTags, class... IndexTypes>
+KOKKOS_FORCEINLINE_FUNCTION typename std::enable_if<
+    sizeof...( IndexTypes ) == FieldTag::rank,
+    typename Particle<FieldTags...>::tuple_type::template member_reference_type<
+        TypeIndexer<FieldTag, FieldTags...>::index>>::type
+get( Particle<FieldTags...>& particle, FieldTag, IndexTypes... indices )
+{
+    return Cabana::get<TypeIndexer<FieldTag, FieldTags...>::index>(
+        particle.tuple(), indices... );
+}
+
+//---------------------------------------------------------------------------//
+//! Return a single SoA for a single particle.
```

Comment:
This returns a single element of a single field via `indices`

Reply:
Fixed